pkg/adaptor/forwarder/forwarder.go

```diff
@@ -134,8 +139,15 @@ func startForwarding(ctx context.Context, shimConn net.Conn, serverURL *url.URL)
 				break
 			}
 
-			logger.Printf("failed to establish an upgraded connection to %s: %v. (retrying...)", serverURL, err)
+			logger.Printf("failed to establish an upgraded connection to %s: %v. (retrying... %d/%d)", serverURL, err, count, maxRetries)
 			<-ctx.Done()
+
+			if count >= maxRetries {
```

Comment:
Not sure if there is a better way to do this. But for now let's go with this and improve later on.

Reply:
Yes. We need a better mechanism to manage mappings between pod and VM.